source/dyaml/node.d

```diff
@@ -2325,9 +2300,16 @@ struct Node
         // This only works for default YAML types, not for user defined types.
         @property bool isType(T)() const
         {
-            return value_.type is typeid(Unqual!T);
+            return value_.match!(
+                (const T _) => true,
+                _    => false,
+            );
         }
 
+        /// Check at compile time if a type is stored natively
+        enum canBeType (T) = is(typeof({ value_.match!((const T _) => true, _ => false); }));
```

Comment:
What's this for? the comment makes it sound identical in purpose to allowed(T)

Reply:
`dstring` will match `allowed` but not `canBeType`.
Before the move to `SumType`, we just relied on `Variant` telling us "no" at runtime, but with `SumType`, we actually hit a `static assert` (a compile-time "no"). To keep the same behavior, I just had to wrap the call to make sure it was only visible to the compiler when `isType` could actually return `true`, not when it wold always return `false`.